pkg/helm/controller/reconcile.go

```diff
@@ -94,38 +96,8 @@ func (r HelmOperatorReconciler) Reconcile(request reconcile.Request) (reconcile.
 	status := types.StatusFor(o)
 	log = log.WithValues("release", manager.ReleaseName())
 
-	deleted := o.GetDeletionTimestamp() != nil
-	pendingFinalizers := o.GetFinalizers()
-	if !deleted && !contains(pendingFinalizers, finalizer) {
-		log.V(1).Info("Adding finalizer", "finalizer", finalizer)
-		finalizers := append(pendingFinalizers, finalizer)
-		o.SetFinalizers(finalizers)
-		err = r.updateResource(o)
-
-		// Need to requeue because finalizer update does not change metadata.generation
-		return reconcile.Result{Requeue: true}, err
-	}
```

Comment:
This section is moved so that we only add the finalizer after a successful install and then we ensure it is present when reconciling.